packages/react/src/components/RadioCardGroup/index.tsx

```diff
@@ -0,0 +1,183 @@
+  name?: string;
+  className?: string;
+  radios: RadioItem[];
+  defaultValue?: string;
+  value?: any;
+  onChange: (radio: RadioItem, input: HTMLElement) => void;
+}
+
+const RadioCardGroup = ({
+  name,
+  radios,
+  defaultValue,
+  value,
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  onChange = () => {},
+  className,
+  ...other
+}: RadioCardGroupProps) => {
+  const [currentValue, setCurrentValue] = useState<string | null>(
+    value || defaultValue || null
+  );
+  const [focusIndex, setFocusIndex] = useState<number | null>(null);
+  const inputs = useRef<HTMLInputElement[]>([]);
+  const handleChange = (value: any) => setCurrentValue(value);
+  const onRadioFocus = (index: number) => setFocusIndex(index);
+  const onRadioBlur = () => setFocusIndex(null);
+  const onRadioClick = (index: number) => {
+    const radio = inputs.current?.[index];
+
+    if (!radio) {
+      return;
+    }
+
+    radio.click();
+    radio.focus();
+  };
+
+  useEffect(() => {
+    if (typeof value === 'undefined') {
+      return;
+    }
+
+    setCurrentValue(value);
+  }, [value]);
+
+  const radioButtons = radios.map((radio, index) => {
+    const {
+      label,
+      disabled,
+      value: radioValue,
+      cardImgSrc,
+      cardIcon,
+      id,
+      className,
+      ...other
+    } = radio;
+    const isChecked = currentValue === radioValue;
+    const isFocused = focusIndex === index;
+
+    return (
+      <div className={classNames('Radio')} key={id}>
+        <Card
+          variant="simple"
+          className={classNames('Radio_card Radio__overlay', {
+            'Radio__overlay--focused': isFocused,
+            'Radio__overlay--checked': isChecked,
+            'Radio__overlay--disabled': disabled
+          })}
+          type={isChecked ? 'radio-checked' : 'radio-unchecked'}
+          onClick={() => onRadioClick(index)}
+        >
+          <input
+            type="radio"
+            name={name}
+            value={radioValue}
+            id={id}
+            ref={input => {
+              if (!input) {
+                return;
+              }
+
+              inputs.current.push(input);
+            }}
+            onFocus={() => onRadioFocus(index)}
+            onBlur={() => onRadioBlur()}
+            onChange={() => {
+              handleChange(radioValue);
+              onChange(radio, inputs.current?.[index]);
+            }}
+            disabled={disabled}
+            checked={isChecked}
+            {...other}
+          />
+          <CardContent>
+            <div className={classNames('Radio_card_icon_div')}>
+              {isChecked && (
+                <Icon
+                  className={classNames('Radio_card--icon')}
+                  type="check-circle"
+                />
+              )}
+            </div>
+
+            <div className={classNames('Radio_card_base')}>
+              <img
+                className={classNames('Radio_card_img')}
```

Comment:
This doesn't follow BEM, a more appropriate name would be something like `RadioCardGroup__Image`